### ISSUE TITLE
Different exceptions with txpool_content rpc call

### ISSUE THREAD

**Issue by mmhh1910** [NethermindEth/nethermind#1180]
Using docker image "nethermind/nethermind:1.3.6" I got this:

```
2019-12-19 09:33:27.0780|Executing JSON RPC call txpool_content with params |

2019-12-19 09:33:27.1241|BlockchainProcessor encountered an exception.|System.NullReferenceException: Object reference not set to an instance of an object.
   at Nethermind.Store.StateProvider.Commit(IReleaseSpec releaseSpec, IStateTracer stateTracer) in /nethermind/src/Nethermind/Nethermind.Store/StateProvider.cs:line 391
   at Nethermind.Evm.TransactionProcessor.Execute(Transaction transaction, BlockHeader block, ITxTracer txTracer, Boolean readOnly) in /nethermind/src/Nethermind/Nethermind.Evm/TransactionProcessor.cs:line 288
   at Nethermind.Blockchain.BlockProcessor.ProcessTransactions(Block block, ProcessingOptions processingOptions, IBlockTracer blockTracer) in /nethermind/src/Nethermind/Nethermind.Blockchain/BlockProcessor.cs:line 162
   at Nethermind.Blockchain.BlockProcessor.ProcessOne(Block suggestedBlock, ProcessingOptions options, IBlockTracer blockTracer) in /nethermind/src/Nethermind/Nethermind.Blockchain/BlockProcessor.cs:line 222
   at Nethermind.Blockchain.BlockProcessor.Process(Keccak branchStateRoot, Block[] suggestedBlocks, ProcessingOptions options, IBlockTracer blockTracer) in /nethermind/src/Nethermind/Nethermind.Blockchain/BlockProcessor.cs:line 125
   at Nethermind.Blockchain.BlockchainProcessor.Process(Block suggestedBlock, ProcessingOptions options, IBlockTracer blockTracer) in /nethermind/src/Nethermind/Nethermind.Blockchain/BlockchainProcessor.cs:line 370
   at Nethermind.Blockchain.BlockchainProcessor.RunProcessingLoop() in /nethermind/src/Nethermind/Nethermind.Blockchain/BlockchainProcessor.cs:line 243
   at System.Threading.Tasks.Task.InnerInvoke()
   at System.Threading.Tasks.Task.<>c.<.cctor>b__274_0(Object obj)
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state)
--- End of stack trace from previous location where exception was thrown ---
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state)
   at System.Threading.Tasks.Task.ExecuteWithThreadLocal(Task& currentTaskSlot, Thread threadPoolThread)

   
2019-12-19 09:33:27.2878|Executing JSON RPC call txpool_content with params |

2019-12-19 09:33:27.2916|Error during method execution, request: ID 1385, txpool_content()|System.NullReferenceException: Object reference not set to an instance of an object.
   at Nethermind.Store.StateProvider.GetThroughCache(Address address) in /nethermind/src/Nethermind/Nethermind.Store/StateProvider.cs:line 571
   at Nethermind.Store.StateProvider.GetNonce(Address address) in /nethermind/src/Nethermind/Nethermind.Store/StateProvider.cs:line 114
   at Nethermind.Blockchain.TxPools.TxPoolInfoProvider.GetInfo() in /nethermind/src/Nethermind/Nethermind.Blockchain/TxPools/TxPoolInfoProvider.cs:line 50
   at Nethermind.JsonRpc.Modules.TxPool.TxPoolModule.txpool_content() in /nethermind/src/Nethermind/Nethermind.JsonRpc/Modules/TxPool/TxPoolModule.cs:line 42

2019-12-19 09:33:27.3087|Error when handling ID 1385, txpool_content() | {"id":1385,"jsonrpc":"2.0","error":{"code":-32603,"message":"System.NullReferenceException: Object reference not set to an instance of an object.
   at Nethermind.Store.StateProvider.GetThroughCache(Address address) in /nethermind/src/Nethermind/Nethermind.Store/StateProvider.cs:line 571
   at Nethermind.Store.StateProvider.GetNonce(Address address) in /nethermind/src/Nethermind/Nethermind.Store/StateProvider.cs:line 114
   at Nethermind.Blockchain.TxPools.TxPoolInfoProvider.GetInfo() in /nethermind/src/Nethermind/Nethermind.Blockchain/TxPools/TxPoolInfoProvider.cs:line 50
   at Nethermind.JsonRpc.Modules.TxPool.TxPoolModule.txpool_content() in /nethermind/src/Nethermind/Nethermind.JsonRpc/Modules/TxPool/TxPoolModule.cs:line 42"}}|
```

**Comment by mmhh1910** [comment]
FYI: I'm kind of calling txpool_content constantly and in parallel. 

Here's another instance with a different/deeper call stack and error message:

```
2019-12-19 11:35:47.1549|Executing JSON RPC call txpool_content with params |
2019-12-19 11:35:47.1549|Error during method execution, request: ID 26, txpool_content()|Nethermind.Store.StateException: Unable to resolve node 0xf90211a061313b963b2879c3a3a8ca96ebd397096b37f5baef5f671aa822548d
 ---> System.IndexOutOfRangeException: Index was outside the bounds of the array.
   at Nethermind.Core.Encoding.RlpStream.ReadByte() in /nethermind/src/Nethermind/Nethermind.Core/Encoding/RlpStream.cs:line 583
   at Nethermind.Core.Encoding.RlpStream.ReadSequenceLength() in /nethermind/src/Nethermind/Nethermind.Core/Encoding/RlpStream.cs:line 540
   at Nethermind.Store.TrieNode.ResolveNode(PatriciaTree tree, Boolean allowCaching) in /nethermind/src/Nethermind/Nethermind.Store/TrieNode.cs:line 213
   --- End of inner exception stack trace ---
   at Nethermind.Store.TrieNode.ResolveNode(PatriciaTree tree, Boolean allowCaching) in /nethermind/src/Nethermind/Nethermind.Store/TrieNode.cs:line 213
   at Nethermind.Store.PatriciaTree.TraverseBranch(TrieNode node, TraverseContext traverseContext) in /nethermind/src/Nethermind/Nethermind.Store/PatriciaTree.cs:line 436
   at Nethermind.Store.PatriciaTree.TraverseNode(TrieNode node, TraverseContext traverseContext) in /nethermind/src/Nethermind/Nethermind.Store/PatriciaTree.cs:line 322
   at Nethermind.Store.PatriciaTree.TraverseBranch(TrieNode node, TraverseContext traverseContext) in /nethermind/src/Nethermind/Nethermind.Store/PatriciaTree.cs:line 436
   at Nethermind.Store.PatriciaTree.TraverseNode(TrieNode node, TraverseContext traverseContext) in /nethermind/src/Nethermind/Nethermind.Store/PatriciaTree.cs:line 322
   at Nethermind.Store.PatriciaTree.TraverseBranch(TrieNode node, TraverseContext traverseContext) in /nethermind/src/Nethermind/Nethermind.Store/PatriciaTree.cs:line 436
   at Nethermind.Store.PatriciaTree.TraverseNode(TrieNode node, TraverseContext traverseContext) in /nethermind/src/Nethermind/Nethermind.Store/PatriciaTree.cs:line 322
   at Nethermind.Store.PatriciaTree.Run(Span`1 updatePath, Int32 nibblesCount, Byte[] updateValue, Boolean isUpdate, Boolean ignoreMissingDelete, Keccak rootHash) in /nethermind/src/Nethermind/Nethermind.Store/PatriciaTree.cs:line 301
   at Nethermind.Store.PatriciaTree.Get(Span`1 rawKey, Keccak rootHash) in /nethermind/src/Nethermind/Nethermind.Store/PatriciaTree.cs:line 224
   at Nethermind.Store.StateProvider.GetState(Address address) in /nethermind/src/Nethermind/Nethermind.Store/StateProvider.cs:line 547
   at Nethermind.Store.StateProvider.GetAndAddToCache(Address address) in /nethermind/src/Nethermind/Nethermind.Store/StateProvider.cs:line 566
   at Nethermind.Store.StateProvider.GetNonce(Address address) in /nethermind/src/Nethermind/Nethermind.Store/StateProvider.cs:line 114
   at Nethermind.Blockchain.TxPools.TxPoolInfoProvider.GetInfo() in /nethermind/src/Nethermind/Nethermind.Blockchain/TxPools/TxPoolInfoProvider.cs:line 51
   at Nethermind.JsonRpc.Modules.TxPool.TxPoolModule.txpool_content() in /nethermind/src/Nethermind/Nethermind.JsonRpc/Modules/TxPool/TxPoolModule.cs:line 42
```

**Comment by mmhh1910** [comment]
And yet a slightly different one:

```
2019-12-19 11:31:17.6817|Executing JSON RPC call txpool_content with params |
2019-12-19 11:31:17.6940|Error during method execution, request: ID 1, txpool_content()|Nethermind.Store.StateException: Unable to resolve node 0x5e1f9b4ac395136672d7fa21772131e8ec9d43f2b14ed7e44ce16a14a7caad3f
 ---> Nethermind.Core.Encoding.RlpException: Expected a sequence prefix to be in the range of <192, 255> and got 55 at position 21 in the message of length 532 starting with f90211a0253c2c833d07ec99025f4f8d01a8c40737b94b3c664a092293eb077edf9558e9a0216432fade84c8c07fadb4dffb6bb7cb60d5d5a42de60380a7f7f0eb7f3b74cba062d6429e95843574845b2f2a2c3022a36f0accd7d1990566936e7400953e09e1a072d2e12005d66f0ffd405a78b5c8687a8f22aa69be281156ad14ef3225e9f6b1a0519510ce102438904561d09f08abe850ccda4038b88d1ac577460e1ebfeaa4bfa05b76a51143a97806f91d83bbd63e4970f14dad4a7b75a0f91c2d9b198b0e8554a0038bb5817cbe37704711ab45e842d4c48b3b30c01ddd7ea783ee25ee65e35fb2a004ed2c8bb7c7d519a19435eea563ad0d3610e26d24490d41a961870687843735a0abeb8a5b75fcab65f8d81567d95eb2be2da555e08f089a3a97757bd690725489a01fca172831f6fe9cfe45e8ff56d9019bdc09170379c3ca32ff5b93550e2fa31ba0184d35967cb6dcfee6ec93fe01ebb455a2f617e1bd7388dff5cd992390f8dbcba05f1b21d742d78b013d1f95eadad962ea259fbf468e99c74804389742f293be5fa092164cfe05418fb2cdf73fc3ba33a2df7ca890cf439fdf2588dad4b043f612a2a03c32e81a2f93c4107f1d8c241ca59a7b6342bd15692de4fe2821c0466bef7522a0b5fbc221e45db929cf863327e6fa59c46898e0b31465bd2f9b1a90643f1a36c3a0553c32f7775974626426204a19214d75ddacf8ac246e5170706f19ed95b758f980
   at Nethermind.Core.Encoding.RlpStream.ReadSequenceLength() in /nethermind/src/Nethermind/Nethermind.Core/Encoding/RlpStream.cs:line 540
   at Nethermind.Store.TrieNode.ResolveNode(PatriciaTree tree, Boolean allowCaching) in /nethermind/src/Nethermind/Nethermind.Store/TrieNode.cs:line 213
   --- End of inner exception stack trace ---
   at Nethermind.Store.TrieNode.ResolveNode(PatriciaTree tree, Boolean allowCaching) in /nethermind/src/Nethermind/Nethermind.Store/TrieNode.cs:line 213
   at Nethermind.Store.PatriciaTree.Run(Span`1 updatePath, Int32 nibblesCount, Byte[] updateValue, Boolean isUpdate, Boolean ignoreMissingDelete, Keccak rootHash) in /nethermind/src/Nethermind/Nethermind.Store/PatriciaTree.cs:line 301
   at Nethermind.Store.PatriciaTree.Get(Span`1 rawKey, Keccak rootHash) in /nethermind/src/Nethermind/Nethermind.Store/PatriciaTree.cs:line 224
   at Nethermind.Store.StateProvider.GetState(Address address) in /nethermind/src/Nethermind/Nethermind.Store/StateProvider.cs:line 547
   at Nethermind.Store.StateProvider.GetAndAddToCache(Address address) in /nethermind/src/Nethermind/Nethermind.Store/StateProvider.cs:line 566
   at Nethermind.Store.StateProvider.GetNonce(Address address) in /nethermind/src/Nethermind/Nethermind.Store/StateProvider.cs:line 114
   at Nethermind.Blockchain.TxPools.TxPoolInfoProvider.GetInfo() in /nethermind/src/Nethermind/Nethermind.Blockchain/TxPools/TxPoolInfoProvider.cs:line 51
   at Nethermind.JsonRpc.Modules.TxPool.TxPoolModule.txpool_content() in /nethermind/src/Nethermind/Nethermind.JsonRpc/Modules/TxPool/TxPoolModule.cs:line 42
2019-12-19 11:31:17.6920|Error during method execution, request: ID 1, txpool_content()|Nethermind.Store.StateException: Unable to resolve node 0x5e1f9b4ac395136672d7fa21772131e8ec9d43f2b14ed7e44ce16a14a7caad3f
```

**Comment by tkstanczak** [comment]
@LukaszRozmej this should be a one liner fix - marking it as not read only in the json rpc method attribute

**Comment by tkstanczak** [comment]
@mmhh1910 I will be reviewing this once again but would love to hear from you if you had success with the recent version of Nethermind

the amount of changes between 1.3.6 and 1.6.25 was quite significant and much of these targeted RPC

**Comment by mmhh1910** [comment]
I'm now syncing again and will retest over the weekend.

**Comment by mmhh1910** [comment]
I can't reproduce the problem with "nethermind/nethermind:1.6.25" any longer.

**Comment by mmhh1910** [comment]
I was too fast with the close. I got the problem again:

```
2020-03-15 14:21:24.7307|Executing JSON RPC call txpool_content with params  ,
2020-03-15 14:21:24.7307|Error during method execution, request: ID 26, txpool_content() System.IndexOutOfRangeException: Index was outside the bounds of the array.,
   at Nethermind.Serialization.Rlp.RlpStream.ReadByte() in /src/Nethermind/Nethermind.Serialization.Rlp/RlpStream.cs:line 590,
   at Nethermind.Serialization.Rlp.RlpStream.ReadSequenceLength() in /src/Nethermind/Nethermind.Serialization.Rlp/RlpStream.cs:line 547,
   at Nethermind.Trie.TrieNode.ResolveNode(PatriciaTree tree, Boolean allowCaching) in /src/Nethermind/Nethermind.Trie/TrieNode.cs:line 245,
   at Nethermind.Trie.PatriciaTree.TraverseBranch(TrieNode node, TraverseContext traverseContext) in /src/Nethermind/Nethermind.Trie/PatriciaTree.cs:line 451,
   at Nethermind.Trie.PatriciaTree.TraverseNode(TrieNode node, TraverseContext traverseContext) in /src/Nethermind/Nethermind.Trie/PatriciaTree.cs:line 337,
   at Nethermind.Trie.PatriciaTree.TraverseBranch(TrieNode node, TraverseContext traverseContext) in /src/Nethermind/Nethermind.Trie/PatriciaTree.cs:line 451,
   at Nethermind.Trie.PatriciaTree.TraverseNode(TrieNode node, TraverseContext traverseContext) in /src/Nethermind/Nethermind.Trie/PatriciaTree.cs:line 337,
   at Nethermind.Trie.PatriciaTree.TraverseBranch(TrieNode node, TraverseContext traverseContext) in /src/Nethermind/Nethermind.Trie/PatriciaTree.cs:line 451,
   at Nethermind.Trie.PatriciaTree.TraverseNode(TrieNode node, TraverseContext traverseContext) in /src/Nethermind/Nethermind.Trie/PatriciaTree.cs:line 337,
   at Nethermind.Trie.PatriciaTree.TraverseBranch(TrieNode node, TraverseContext traverseContext) in /src/Nethermind/Nethermind.Trie/PatriciaTree.cs:line 451,
   at Nethermind.Trie.PatriciaTree.TraverseNode(TrieNode node, TraverseContext traverseContext) in /src/Nethermind/Nethermind.Trie/PatriciaTree.cs:line 337,
   at Nethermind.Trie.PatriciaTree.TraverseBranch(TrieNode node, TraverseContext traverseContext) in /src/Nethermind/Nethermind.Trie/PatriciaTree.cs:line 451,
   at Nethermind.Trie.PatriciaTree.TraverseNode(TrieNode node, TraverseContext traverseContext) in /src/Nethermind/Nethermind.Trie/PatriciaTree.cs:line 337,
   at Nethermind.Trie.PatriciaTree.TraverseBranch(TrieNode node, TraverseContext traverseContext) in /src/Nethermind/Nethermind.Trie/PatriciaTree.cs:line 451,
   at Nethermind.Trie.PatriciaTree.TraverseNode(TrieNode node, TraverseContext traverseContext) in /src/Nethermind/Nethermind.Trie/PatriciaTree.cs:line 337,
   at Nethermind.Trie.PatriciaTree.TraverseBranch(TrieNode node, TraverseContext traverseContext) in /src/Nethermind/Nethermind.Trie/PatriciaTree.cs:line 451,
   at Nethermind.Trie.PatriciaTree.TraverseNode(TrieNode node, TraverseContext traverseContext) in /src/Nethermind/Nethermind.Trie/PatriciaTree.cs:line 337,
   at Nethermind.Trie.PatriciaTree.Run(Span`1 updatePath, Int32 nibblesCount, Byte[] updateValue, Boolean isUpdate, Boolean ignoreMissingDelete, Keccak rootHash) in /src/Nethermind/Nethermind.Trie/PatriciaTree.cs:line 316,
   at Nethermind.Trie.PatriciaTree.Get(Span`1 rawKey, Keccak rootHash) in /src/Nethermind/Nethermind.Trie/PatriciaTree.cs:line 227,
   at Nethermind.State.StateProvider.GetState(Address address) in /src/Nethermind/Nethermind.State/StateProvider.cs:line 608,
   at Nethermind.State.StateProvider.GetAndAddToCache(Address address) in /src/Nethermind/Nethermind.State/StateProvider.cs:line 635,
   at Nethermind.State.StateProvider.GetNonce(Address address) in /src/Nethermind/Nethermind.State/StateProvider.cs:line 136,
   at Nethermind.TxPool.TxPoolInfoProvider.GetInfo() in /src/Nethermind/Nethermind.TxPool/TxPoolInfoProvider.cs:line 51,
   at Nethermind.JsonRpc.Modules.TxPool.TxPoolModule.txpool_content() in /src/Nethermind/Nethermind.JsonRpc/Modules/TxPool/TxPoolModule.cs:line 42,
2020-03-15 14:21:24.7307|Error when handling ID 26, txpool_content() | {"jsonrpc":"2.0","error":{"code":-32603,"message":"Internal error","data":"System.IndexOutOfRangeException: Index was outside the bounds of the array.\n   at Nethermind.Serialization.Rlp.RlpStream.ReadByte() in /src/Nethermind/Nethermind.Serialization.Rlp/RlpStream.cs:line 590\n   at Nethermind.Serialization.Rlp.RlpStream.ReadSequenceLength() in /src/Nethermind/Nethermind.Serialization.Rlp/RlpStream.cs:line 547\n   at Nethermind.Trie.TrieNode.ResolveNode(PatriciaTree tree, Boolean allowCaching) in /src/Nethermind/Nethermind.Trie/TrieNode.cs:line 245\n   at Nethermind.Trie.PatriciaTree.TraverseBranch(TrieNode node, TraverseContext traverseContext) in /src/Nethermind/Nethermind.Trie/PatriciaTree.cs:line 451\n   at Nethermind.Trie.PatriciaTree.TraverseNode(TrieNode node, TraverseContext traverseContext) in /src/Nethermind/Nethermind.Trie/PatriciaTree.cs:line 337\n   at Nethermind.Trie.PatriciaTree.TraverseBranch(TrieNode node, TraverseContext traverseContext) in /src/Nethermind/Nethermind.Trie/PatriciaTree.cs:line 451\n   at Nethermind.Trie.PatriciaTree.TraverseNode(TrieNode node, TraverseContext traverseContext) in /src/Nethermind/Nethermind.Trie/PatriciaTree.cs:line 337\n   at Nethermind.Trie.PatriciaTree.TraverseBranch(TrieNode node, TraverseContext traverseContext) in /src/Nethermind/Nethermind.Trie/PatriciaTree.cs:line 451\n   at Nethermind.Trie.PatriciaTree.TraverseNode(TrieNode node, TraverseContext traverseContext) in /src/Nethermind/Nethermind.Trie/PatriciaTree.cs:line 337\n   at Nethermind.Trie.PatriciaTree.TraverseBranch(TrieNode node, TraverseContext traverseContext) in /src/Nethermind/Nethermind.Trie/PatriciaTree.cs:line 451\n   at Nethermind.Trie.PatriciaTree.TraverseNode(TrieNode node, TraverseContext traverseContext) in /src/Nethermind/Nethermind.Trie/PatriciaTree.cs:line 337\n   at Nethermind.Trie.PatriciaTree.TraverseBranch(TrieNode node, TraverseContext traverseContext) in /src/Nethermind/Nethermind.Trie/PatriciaTree.cs:line 451\n   at Nethermind.Trie.PatriciaTree.TraverseNode(TrieNode node, TraverseContext traverseContext) in /src/Nethermind/Nethermind.Trie/PatriciaTree.cs:line 337\n   at Nethermind.Trie.PatriciaTree.TraverseBranch(TrieNode node, TraverseContext traverseContext) in /src/Nethermind/Nethermind.Trie/PatriciaTree.cs:line 451\n   at Nethermind.Trie.PatriciaTree.TraverseNode(TrieNode node, TraverseContext traverseContext) in /src/Nethermind/Nethermind.Trie/PatriciaTree.cs:line 337\n   at Nethermind.Trie.PatriciaTree.TraverseBranch(TrieNode node, TraverseContext traverseContext) in /src/Nethermind/Nethermind.Trie/PatriciaTree.cs:line 451\n   at Nethermind.Trie.PatriciaTree.TraverseNode(TrieNode node, TraverseContext traverseContext) in /src/Nethermind/Nethermind.Trie/PatriciaTree.cs:line 337\n   at Nethermind.Trie.PatriciaTree.Run(Span`1 updatePath, Int32 nibblesCount, Byte[] updateValue, Boolean isUpdate, Boolean ignoreMissingDelete, Keccak rootHash) in /src/Nethermind/Nethermind.Trie/PatriciaTree.cs:line 316\n   at Nethermind.Trie.PatriciaTree.Get(Span`1 rawKey, Keccak rootHash) in /src/Nethermind/Nethermind.Trie/PatriciaTree.cs:line 227\n   at Nethermind.State.StateProvider.GetState(Address address) in /src/Nethermind/Nethermind.State/StateProvider.cs:line 608\n   at Nethermind.State.StateProvider.GetAndAddToCache(Address address) in /src/Nethermind/Nethermind.State/StateProvider.cs:line 635\n   at Nethermind.State.StateProvider.GetNonce(Address address) in /src/Nethermind/Nethermind.State/StateProvider.cs:line 136\n   at Nethermind.TxPool.TxPoolInfoProvider.GetInfo() in /src/Nethermind/Nethermind.TxPool/TxPoolInfoProvider.cs:line 51\n   at Nethermind.JsonRpc.Modules.TxPool.TxPoolModule.txpool_content() in /src/Nethermind/Nethermind.JsonRpc/Modules/TxPool/TxPoolModule.cs:line 42"},"id":26} ,
2020-03-15 14:21:24.7307|Error during method execution, request: ID 27, txpool_content() System.IndexOutOfRangeException: Index was outside the bounds of the array.,
   at Nethermind.Serialization.Rlp.RlpStream.ReadByte() in /src/Nethermind/Nethermind.Serialization.Rlp/RlpStream.cs:line 590,
   at Nethermind.Serialization.Rlp.RlpStream.ReadSequenceLength() in /src/Nethermind/Nethermind.Serialization.Rlp/RlpStream.cs:line 547,
   at Nethermind.Trie.TrieNode.ResolveNode(PatriciaTree tree, Boolean allowCaching) in /src/Nethermind/Nethermind.Trie/TrieNode.cs:line 245,
   at Nethermind.Trie.PatriciaTree.TraverseBranch(TrieNode node, TraverseContext traverseContext) in /src/Nethermind/Nethermind.Trie/PatriciaTree.cs:line 451,
   at Nethermind.Trie.PatriciaTree.TraverseNode(TrieNode node, TraverseContext traverseContext) in /src/Nethermind/Nethermind.Trie/PatriciaTree.cs:line 337,
   at Nethermind.Trie.PatriciaTree.TraverseBranch(TrieNode node, TraverseContext traverseContext) in /src/Nethermind/Nethermind.Trie/PatriciaTree.cs:line 451,
   at Nethermind.Trie.PatriciaTree.TraverseNode(TrieNode node, TraverseContext traverseContext) in /src/Nethermind/Nethermind.Trie/PatriciaTree.cs:line 337,
   at Nethermind.Trie.PatriciaTree.TraverseBranch(TrieNode node, TraverseContext traverseContext) in /src/Nethermind/Nethermind.Trie/PatriciaTree.cs:line 451,
   at Nethermind.Trie.PatriciaTree.TraverseNode(TrieNode node, TraverseContext traverseContext) in /src/Nethermind/Nethermind.Trie/PatriciaTree.cs:line 337,
   at Nethermind.Trie.PatriciaTree.TraverseBranch(TrieNode node, TraverseContext traverseContext) in /src/Nethermind/Nethermind.Trie/PatriciaTree.cs:line 451,
   at Nethermind.Trie.PatriciaTree.TraverseNode(TrieNode node, TraverseContext traverseContext) in /src/Nethermind/Nethermind.Trie/PatriciaTree.cs:line 337,
   at Nethermind.Trie.PatriciaTree.TraverseBranch(TrieNode node, TraverseContext traverseContext) in /src/Nethermind/Nethermind.Trie/PatriciaTree.cs:line 451,
   at Nethermind.Trie.PatriciaTree.TraverseNode(TrieNode node, TraverseContext traverseContext) in /src/Nethermind/Nethermind.Trie/PatriciaTree.cs:line 337,
   at Nethermind.Trie.PatriciaTree.TraverseBranch(TrieNode node, TraverseContext traverseContext) in /src/Nethermind/Nethermind.Trie/PatriciaTree.cs:line 451,
   at Nethermind.Trie.PatriciaTree.TraverseNode(TrieNode node, TraverseContext traverseContext) in /src/Nethermind/Nethermind.Trie/PatriciaTree.cs:line 337,
   at Nethermind.Trie.PatriciaTree.TraverseBranch(TrieNode node, TraverseContext traverseContext) in /src/Nethermind/Nethermind.Trie/PatriciaTree.cs:line 451,
   at Nethermind.Trie.PatriciaTree.TraverseNode(TrieNode node, TraverseContext traverseContext) in /src/Nethermind/Nethermind.Trie/PatriciaTree.cs:line 337,
   at Nethermind.Trie.PatriciaTree.Run(Span`1 updatePath, Int32 nibblesCount, Byte[] updateValue, Boolean isUpdate, Boolean ignoreMissingDelete, Keccak rootHash) in /src/Nethermind/Nethermind.Trie/PatriciaTree.cs:line 316,
   at Nethermind.Trie.PatriciaTree.Get(Span`1 rawKey, Keccak rootHash) in /src/Nethermind/Nethermind.Trie/PatriciaTree.cs:line 227,
   at Nethermind.State.StateProvider.GetState(Address address) in /src/Nethermind/Nethermind.State/StateProvider.cs:line 608,
   at Nethermind.State.StateProvider.GetAndAddToCache(Address address) in /src/Nethermind/Nethermind.State/StateProvider.cs:line 635,
   at Nethermind.State.StateProvider.GetNonce(Address address) in /src/Nethermind/Nethermind.State/StateProvider.cs:line 136,
   at Nethermind.TxPool.TxPoolInfoProvider.GetInfo() in /src/Nethermind/Nethermind.TxPool/TxPoolInfoProvider.cs:line 51,
   at Nethermind.JsonRpc.Modules.TxPool.TxPoolModule.txpool_content() in /src/Nethermind/Nethermind.JsonRpc/Modules/TxPool/TxPoolModule.cs:line 42,
2020-03-15 14:21:24.7320|Error when handling ID 27, txpool_content() | {"jsonrpc":"2.0","error":{"code":-32603,"message":"Internal error","data":"System.IndexOutOfRangeException: Index was outside the bounds of the array.\n   at Nethermind.Serialization.Rlp.RlpStream.ReadByte() in /src/Nethermind/Nethermind.Serialization.Rlp/RlpStream.cs:line 590\n   at Nethermind.Serialization.Rlp.RlpStream.ReadSequenceLength() in /src/Nethermind/Nethermind.Serialization.Rlp/RlpStream.cs:line 547\n   at Nethermind.Trie.TrieNode.ResolveNode(PatriciaTree tree, Boolean allowCaching) in /src/Nethermind/Nethermind.Trie/TrieNode.cs:line 245\n   at Nethermind.Trie.PatriciaTree.TraverseBranch(TrieNode node, TraverseContext traverseContext) in /src/Nethermind/Nethermind.Trie/PatriciaTree.cs:line 451\n   at Nethermind.Trie.PatriciaTree.TraverseNode(TrieNode node, TraverseContext traverseContext) in /src/Nethermind/Nethermind.Trie/PatriciaTree.cs:line 337\n   at Nethermind.Trie.PatriciaTree.TraverseBranch(TrieNode node, TraverseContext traverseContext) in /src/Nethermind/Nethermind.Trie/PatriciaTree.cs:line 451\n   at Nethermind.Trie.PatriciaTree.TraverseNode(TrieNode node, TraverseContext traverseContext) in /src/Nethermind/Nethermind.Trie/PatriciaTree.cs:line 337\n   at Nethermind.Trie.PatriciaTree.TraverseBranch(TrieNode node, TraverseContext traverseContext) in /src/Nethermind/Nethermind.Trie/PatriciaTree.cs:line 451\n   at Nethermind.Trie.PatriciaTree.TraverseNode(TrieNode node, TraverseContext traverseContext) in /src/Nethermind/Nethermind.Trie/PatriciaTree.cs:line 337\n   at Nethermind.Trie.PatriciaTree.TraverseBranch(TrieNode node, TraverseContext traverseContext) in /src/Nethermind/Nethermind.Trie/PatriciaTree.cs:line 451\n   at Nethermind.Trie.PatriciaTree.TraverseNode(TrieNode node, TraverseContext traverseContext) in /src/Nethermind/Nethermind.Trie/PatriciaTree.cs:line 337\n   at Nethermind.Trie.PatriciaTree.TraverseBranch(TrieNode node, TraverseContext traverseContext) in /src/Nethermind/Nethermind.Trie/PatriciaTree.cs:line 451\n   at Nethermind.Trie.PatriciaTree.TraverseNode(TrieNode node, TraverseContext traverseContext) in /src/Nethermind/Nethermind.Trie/PatriciaTree.cs:line 337\n   at Nethermind.Trie.PatriciaTree.TraverseBranch(TrieNode node, TraverseContext traverseContext) in /src/Nethermind/Nethermind.Trie/PatriciaTree.cs:line 451\n   at Nethermind.Trie.PatriciaTree.TraverseNode(TrieNode node, TraverseContext traverseContext) in /src/Nethermind/Nethermind.Trie/PatriciaTree.cs:line 337\n   at Nethermind.Trie.PatriciaTree.TraverseBranch(TrieNode node, TraverseContext traverseContext) in /src/Nethermind/Nethermind.Trie/PatriciaTree.cs:line 451\n   at Nethermind.Trie.PatriciaTree.TraverseNode(TrieNode node, TraverseContext traverseContext) in /src/Nethermind/Nethermind.Trie/PatriciaTree.cs:line 337\n   at Nethermind.Trie.PatriciaTree.Run(Span`1 updatePath, Int32 nibblesCount, Byte[] updateValue, Boolean isUpdate, Boolean ignoreMissingDelete, Keccak rootHash) in /src/Nethermind/Nethermind.Trie/PatriciaTree.cs:line 316\n   at Nethermind.Trie.PatriciaTree.Get(Span`1 rawKey, Keccak rootHash) in /src/Nethermind/Nethermind.Trie/PatriciaTree.cs:line 227\n   at Nethermind.State.StateProvider.GetState(Address address) in /src/Nethermind/Nethermind.State/StateProvider.cs:line 608\n   at Nethermind.State.StateProvider.GetAndAddToCache(Address address) in /src/Nethermind/Nethermind.State/StateProvider.cs:line 635\n   at Nethermind.State.StateProvider.GetNonce(Address address) in /src/Nethermind/Nethermind.State/StateProvider.cs:line 136\n   at Nethermind.TxPool.TxPoolInfoProvider.GetInfo() in /src/Nethermind/Nethermind.TxPool/TxPoolInfoProvider.cs:line 51\n   at Nethermind.JsonRpc.Modules.TxPool.TxPoolModule.txpool_content() in /src/Nethermind/Nethermind.JsonRpc/Modules/TxPool/TxPoolModule.cs:line 42"},"id":27} ,
2020-03-15 14:21:24.7320|Executing JSON RPC call txpool_content with params  ,
```

**Comment by tkstanczak** [comment]
https://github.com/NethermindEth/nethermind/pull/1547

**Comment by tkstanczak** [comment]
@mmhh1910 the potential (very likely to work) fix has been included in 1.6.31 release

**Comment by mmhh1910** [comment]
Can't reproduce any longer with docker image nethermind/nethermind:1.7.3.
